### PR TITLE
[FEAT] exp expression implementation

### LINKED ISSUE #1909: [EXPRESSIONS] `.exp`
`.exp()`

Compute  e^self

### PR DESCRIPTION
**Pull request by MeepoWin**
This closes #1909.

### CODE REVIEW

File: tests/table/test_eval.py [hunk 1]
```diff
@@ -266,3 +266,9 @@ def test_table_round_bad_input() -> None:
 
     with pytest.raises(ValueError, match="decimal can not be negative: -2"):
         table.eval_expression_list([col("a").round(-2)])
+
+
+def test_table_exp() -> None:
```
Comment (samster25) [inline tests/table/test_eval.py:271]: we should also add a test on an integer series

**Review by samster25 — APPROVED** [review]
Looks great! Let just add a bit more testing :)

**Comment by MeepoWin** [comment]
Thanks @samster25 for the quick review! I have added the test and rebased on mater to resolve the conflict.

**Comment by samster25** [comment]
Thanks @MeepoWin for the quick turnaround!

**Pull request merged**